bokehjs/src/lib/api/gridplot.ts

```diff
@@ -85,14 +87,28 @@ export function group_tools(tools: ToolLike<Tool>[], merge?: MergeFn,
   return computed
 }
 
-export function gridplot(children: (LayoutDOM | null)[][] | Matrix<LayoutDOM | null>, options: GridPlotOpts = {}): GridPlot {
+export function gridplot(children: (UIElement | null)[], options: GridPlotOpts & {ncols: number}): GridPlot
+export function gridplot(children: (UIElement | null)[][], options?: GridPlotOpts): GridPlot
+
+export function gridplot(children: (UIElement | null)[] | (UIElement | null)[][] | Matrix<UIElement | null>, options: GridPlotOpts & {ncols?: number} = {}): GridPlot {
   const toolbar_location = options.toolbar_location
   const merge_tools      = options.merge_tools ?? true
   const sizing_mode      = options.sizing_mode
+  const ncols            = options.ncols
+
+  const matrix = (() => {
+    const has_array = isArrayOf(children, (c) => c == null || c instanceof UIElement)
```

Comment:
```suggestion
    const has_array = isArrayOf(children, (c) => c == null || c instanceof UIElement)
```

I'm not entirely sure why `c == null` will result in `has_array`.

Reply:
`null` indicates an empty spot. `gridplot()` takes either an array of `null | UIElement` or a matrix (array of arrays) of `null | UIElement`.

Reply:
Could this lead to false positives? And should the check be `===` if we only want to check for `null`?

Reply:
As long as we use either double or triple equals consistently then no. Probably it would be better to use `null | undefined` type.